modnet/hyper_opt/fit_genetic.py

```diff
@@ -96,9 +96,11 @@ def crossover(self, partner: Individual) -> Individual:
         )  # creates indices to take randomly half the genes from one parent, and half the genes from the other
 
         child_genes = {
-            list(self.genes.keys())[i]: list(self.genes.values())[i]
-            if i in genes_from_mother
-            else list(partner.genes.values())[i]
+            list(self.genes.keys())[i]: (
+                list(self.genes.values())[i]
+                if i in genes_from_mother
+                else list(partner.genes.values())[i]
+            )
```

Comment:
As it stands this is going to remake the genes lists for every iteration of the dict comprehension, could we tidy it up a bit?

(I know this change is just from auto-formatting)

something like

```python

genes_from_mother = random.sample(sorted(self.genes)), k=len(self.genes) // 2)
child_genes = {key: gene for key in self.genes if key in genes_from_mother else partner.genes[key]}
```

should work and just directly give you the keys to use?

Reply:
Indeed !